apps/finance/app/src/components/Transfers.js

```diff
@@ -29,236 +30,220 @@ const TRANSFER_TYPES = [
-    }))
-  }
+  })
 
-  // Filter transfer based on the selected filters
-  getFilteredTransfers({
+  const handleToggleFiltersClick = React.useCallback(() => {
+    setFiltersOpened(!filtersOpened)
+  })
+  const handleTokenChange = React.useCallback(index => {
+    setSelectedToken(index)
+    setDisplayedTransfers(TRANSFERS_PER_PAGE)
+  })
+  const handleTransferTypeChange = React.useCallback(index => {
+    setSelectedTransferType(index)
+    setDisplayedTransfers(TRANSFERS_PER_PAGE)
+  })
+  const handleResetFilters = React.useCallback(() => {
+    setDisplayedTransfers(TRANSFERS_PER_PAGE)
+    setSelectedToken(0)
+    setSelectedTransferType(0)
+  })
+  const getCsvFilename = React.useCallback(() => {
+    let filename = 'transfers.csv'
+    const { start, end } = selectedDateRange
+    if (start && end) {
+      const formattedStart = format(selectedDateRange.start, 'yyyy-MM-dd')
+      const formattedEnd = format(selectedDateRange.end, 'yyyy-MM-dd')
+      filename = `transfers_${formattedStart}_to_${formattedEnd}.csv`
+    }
+    return filename
+  })
+  const showMoreTransfers = React.useCallback(() => {
+    setDisplayedTransfers(displayedTransfers + TRANSFERS_PER_PAGE)
+  })
+  const filteredTransfers = getFilteredTransfers({
     tokens,
     transactions,
     selectedToken,
     selectedTransferType,
     selectedDateRange,
-  }) {
-    const transferType = TRANSFER_TYPES[selectedTransferType]
-    return transactions.filter(
-      ({ token, isIncoming, date }) =>
-        (!selectedDateRange.start || !selectedDateRange.end ||
-          isWithinInterval(new Date(date), {
-            start: startOfDay(selectedDateRange.start),
-            end: endOfDay(selectedDateRange.end),
-          })) &&
-        (selectedToken === 0 ||
-          addressesEqual(token, tokens[selectedToken - 1].address)) &&
-        (transferType === TransferTypes.All ||
-          (transferType === TransferTypes.Incoming && isIncoming) ||
-          (transferType === TransferTypes.Outgoing && !isIncoming))
+  })
+  const symbols = tokens.map(({ symbol }) => symbol)
+  const tokenDetails = tokens.reduce(reduceTokenDetails, {})
+  const filtersActive = selectedToken !== 0 || selectedTransferType !== 0
+  const { resolve } = React.useContext(IdentityContext)
+  const getDownloadData = async () => {
+    const mappedData = await Promise.all(
+      filteredTransfers.map(
+        async ({
+          date,
+          numData: { amount },
+          reference,
+          isIncoming,
+          entity,
+          token,
+        }) => {
+          const { symbol, decimals } = tokenDetails[toChecksumAddress(token)]
+          const formattedAmount = formatTokenAmount(
+            amount,
+            isIncoming,
+            decimals,
+            true,
+            { rounding: 5 }
+          )
+          const { name = null } = (await resolve(entity)) || {}
+          return `${formatDate(date)},${
+            name ? `${name} (${entity})` : entity
```

Comment:
Perhaps we should use a different column for the `name`, so the data import / manipulation is easier (much easier to annotate existing columns with a name when it's a separate column than for the tool to support text extraction from a field).

Reply:
Added this change as it makes sense to me but cc @luisivan to double check

Reply:
Makes a lot of sense